smarts/core/controllers/lane_following_controller.py

```diff
@@ -78,8 +78,8 @@ def perform_lane_following(
         state = controller_state
         # This lookahead value is coupled with a few calculations below, changing it
         # may affect stability of the controller.
-        wp_paths = sensor_state.mission_planner.waypoint_paths_at(
-            vehicle.pose, lookahead=16
+        wp_paths = sim.road_map.waypoint_paths(
+            vehicle.pose, lookahead=16, route=sensor_state.plan.route
```

Comment:
No, it's the same as it was before.  Passing in the `route` was just made necessary because the `waypoint_path` method was moved out of `Plan` (which used to be `MissionPlan`), where it was a member, and into `RoadMap`, where it's not.  (This was your suggestion, btw!  ...including passing in the route.)  Look at `mission_planner.py` on the current `develop` branch:  the `waypoint_paths_at()` method there takes a `constrain_to_route` parameter, which defaults to `True`.   So in the pre-PR code, this was attempting to constrain to the route, which was (pre-)stored in the `MissionPlanner` object (and accessed in the old `_edge_ids()` method).

Also note that if the `route` object that is passed has an empty `roads` field (as it does in the case of an endless mission), the waypoints don't get filtered either.

Reply:
Ah I see, yes, I missed noticing the mission planner. Of course then it would need the route if it is just maintaining the existing behavior. I think I was just most put off by the fact that the lane following controller follows the route more-so than the lane.